juju/juju
# Support base (os/channel) instead of series in more api calls

Add support in more rpc calls for using base (os/channel) instead of series. For 2.9, both will be supported and series is removed in 3.0. The affected rpc params structs include:
- status
- get application info results
- caas provisioning info
- provisioning info
- multiwatcher info
- image metadata
- backup info

The provisioning params are controller apis only so series can be replaced. For the client api params, both series and channel are supported.

The format for printing bases is changes from "ubuntu/22.04" to "ubuntu:22.04".

The min juju version for upgrades is changed to 2.9.35 to account for these changes.

## Checklist


## QA steps

deploy this bundle
```
applications:
  juju-qa-test:
    charm: juju-qa-test
    channel: stable
    num_units: 2
  nrpe:
    charm: cs:nrpe-64
    channel: stable
  ntp:
    charm: cs:ntp-40
    channel: stable
    series: bionic
  telegraf:
    charm: cs:telegraf-37
    channel: stable
  ubuntu:
    charm: ubuntu
    series: xenial
    num_units: 2
      #    to:
      #      - "lxd:new"
relations:
- - ntp:juju-info
  - juju-qa-test:juju-info
- - ntp:juju-info
  - ubuntu:juju-info
- - nrpe:general-info
  - ubuntu:juju-info
- - telegraf:juju-info
  - ubuntu:juju-info
```

redeploy the same bundle
do the same with a 2.9.34 cli client
run `juju diff-bundle ...` and check the output

repeat the above with this k8s bundle
```
bundle: kubernetes
applications:
  mariadb:
    charm: cs:~juju/mariadb-k8s
    scale: 2
    constraints: mem=1G
    options:
        dataset-size: 70%
    storage:
      database: mariadb-pv,20M
  gitlab:
    charm: cs:~juju/gitlab-k8s
    placement: foo=bar
    scale: 1
relations:
  - - gitlab:mysql
    - mariadb:server
```

`juju status --format yaml` and check that bases are included

Immediately after deploying a bundle (before the image metadata is automatically purged), run `juju metadata list-images`
